homeassistant/components/fritz/device_tracker.py

```diff
@@ -63,35 +63,47 @@ async def async_get_scanner(hass: HomeAssistant, config: ConfigType):
     return None
 
 
+class FritzData:
+    """Storage class for platform global data."""
+
+    def __init__(self) -> None:
+        """Initialize the data."""
+        self.tracked = set()
+
+
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities
 ) -> None:
     """Set up device tracker for FRITZ!Box component."""
     _LOGGER.debug("Starting FRITZ!Box device tracker")
     router = hass.data[DOMAIN][entry.entry_id]
-    tracked = set()
+
+    if DATA_FRITZ not in hass.data:
+        hass.data[DATA_FRITZ] = FritzData()
+
+    data_fritz = hass.data[DATA_FRITZ]
```

Comment:
This needs to be cleaned up when the last fritz box config entry is unloaded. We should probably create this in `__init__.py`

Reply:
`isy994` has an example to see if all config entries are unloaded

Reply:
Also if you remove one router, the ones created by that router should be removed from the set.

Reply:
Thanks @bdraco! That makes sense. I will try to address it.

Reply:
See new commit